cloudify_kubernetes/tasks.py

```diff
@@ -158,6 +158,60 @@ def _do_resource_status_check(resource_kind, response):
         else:
             ctx.logger.debug('status {0}'.format(status))
 
+    elif resource_kind == 'Deployment':
+        conditions = response['status']['conditions']
+        if conditions:
+            for condition in conditions:
+                if condition['type'] == 'Available':
+                    ctx.logger.debug('Deployment condition is Available')
+
+                elif condition['type'] == 'ReplicaFailure':
+                    raise NonRecoverableError(
+                        'Deployment condition is ReplicaFailure ,'
+                        'reason:{0}, message: {1}'
+                        ''.format(condition['reason'], condition['message']))
+
+                elif condition['type'] == 'Progressing':
+                    raise OperationRetry(
+                        'Deployment condition is Progressing')
+        else:
+            raise OperationRetry('Deployment condition is not ready yet')
+
+    elif resource_kind == 'PersistentVolumeClaim':
+        status = response['status']['phase']
+        if status in ['Pending', 'Available']:
+            raise OperationRetry(
+                'PersistentVolumeClaim status is {},'
+                'No persistent volume is detected'.format(status))
+
+        elif status == 'Bound':
+            ctx.logger.debug('PersistentVolumeClaim status is Bound')
+
+    elif resource_kind == 'PersistentVolume':
+        status = response['status']['phase']
+        if status in ['Bound', 'Available']:
+            ctx.logger.debug('PersistentVolume status is {0}'.format(status))
+
+        else:
+            raise OperationRetry(
+                'Unknown PersistentVolume status {0}'.format(status))
+
+    elif resource_kind in ['ReplicaSet', 'ReplicationController']:
+        ready_replicas = response['status'].get('ready_replicas')
+        replicas = response['status']['replicas']
+
```

Comment:
What if ready_replicas should be None? (I don't know if there is such a case.)

Reply:
Good Catch, I will check this as well and update it

Reply:
For ready_replicas it could be None that means no pod was provided and created based on blueprint , so I think it is a valid  option